src/Agent.Worker/CacheDirectoryManager.cs

```diff
@@ -0,0 +1,111 @@
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Net.Http;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    [ServiceLocator(Default = typeof(CacheDirectoryManager))]
+    public interface ICacheDirectoryManager : IAgentService
+    {
+        Task Initialize(IExecutionContext jobContext);
+        Task Cleanup(IExecutionContext jobContext);
+    }
+
+    public sealed class CacheDirectoryManager : AgentService, IJobRunner
+    {
+        public async Task Initialize(IExecutionContext jobContext)
+        {
+            ArgUtil.NotNull(jobContext, nameof(jobContext);
+
+#if OS_WINDOWS
+            // TEMP and TMP.
+            if (ConvertToBoolean(Environment.GetEnvironmentVariable(Constants.EnvironmentVariables.OverrideTemp)) == false)
+            {
+                jobContext.Debug($"Skipping override {Constants.EnvironmentVariables.Temp}");
+            }
+            else
+            {
+                string tempDirectory = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), Constants.Path.TempDirectory);
+                jobContext.Debug($"SET {Constants.EnvironmentVariables.Temp}={tempDirectory}");
+                Directory.CreateDirectory(tempDirectory);
+                SetEnvironmentVariable(Constants.EnvironmentVariables.Temp, tempDirectory);
+                SetEnvironmentVariable(Constants.EnvironmentVariables.Tmp, tempDirectory);
+
+                jobContext.Debug($"Cleaning {Constants.EnvironmentVariables.Temp}");
+                try
+                {
+                    IOUtil.Delete(tempDirectory, contentsOnly: true, cancellationToken: jobContext.CancellationToken);
```

Comment:
the pattern here is, bail on the first error. i wonder if it would be better to pass a new flag `ignoreErrors`?